glib/src/translate.rs

```diff
@@ -2252,6 +2252,49 @@ mod tests {
     use crate::GString;
     use std::collections::HashMap;
 
+    #[test]
+    fn boolean() {
+        assert_eq!(true.to_glib(), ffi::GTRUE);
+        assert_eq!(false.to_glib(), ffi::GFALSE);
+        assert_eq!(true, unsafe { bool::from_glib(ffi::GTRUE) });
+        assert_eq!(false, unsafe { bool::from_glib(ffi::GFALSE) });
+        assert_eq!(true, unsafe { bool::from_glib(42) });
+    }
+
+    #[test]
+    fn ordering() {
+        assert_eq!(Ordering::Less.to_glib(), -1);
+        assert_eq!(Ordering::Equal.to_glib(), 0);
+        assert_eq!(Ordering::Greater.to_glib(), 1);
+        assert_eq!(Ordering::Less, unsafe { Ordering::from_glib(-42) });
+        assert_eq!(Ordering::Less, unsafe { Ordering::from_glib(-1) });
+        assert_eq!(Ordering::Equal, unsafe { Ordering::from_glib(0) });
+        assert_eq!(Ordering::Greater, unsafe { Ordering::from_glib(1) });
+        assert_eq!(Ordering::Greater, unsafe { Ordering::from_glib(42) });
+    }
+
+    #[test]
+    fn string() {
+        let s = "ABC";
+        let owned = "ABC".to_string();
+        let cstring = CString::new("ABC").unwrap();
+
+        let stash = s.to_glib_none();
+        assert_eq!(unsafe { CStr::from_ptr(stash.0) }, cstring.as_c_str());
+
+        let stash = owned.to_glib_none();
+        assert_eq!(unsafe { CStr::from_ptr(stash.0) }, cstring.as_c_str());
+
+        let ptr: *mut c_char = s.to_glib_full();
```

Comment:
You're leaking `ptr` here

Reply:
no, I consume it again in

```
        assert_eq!(owned, unsafe { String::from_glib_full(ptr) });
```

(at least as far as I understand)

Reply:
Yes there you consume the second one. You overwrite the first `ptr`

```rust
        let ptr: *mut c_char = s.to_glib_full();
        [...]
        let ptr: *mut c_char = owned.to_glib_full();
```